src/main/java/simplifier/model/User.java

```diff
@@ -53,4 +55,25 @@ public String getEncryptedPassword() {
   public void setEncryptedPassword(String encryptedPassword) {
     this.encryptedPassword = encryptedPassword;
   }
+
+  @Override
+  public int hashCode() {
+    HashCodeBuilder hcb = new HashCodeBuilder();
+    hcb.append(username);
+    return hcb.toHashCode();
+  }
+
+  @Override
+  public boolean equals(Object obj) {
```

Comment:
The problem is user with same username can be saved and then retrieving 2 users from repository with same usernames but different id and passwords will be equal